### PR TITLE
Chore: Migrate containers/Button to Typescript and Hooks

### PR DESCRIPTION
**Pull request by try-catch-stack**
## Proposed changes

* Migrate to functional component
* Use `useTheme` hook 

## How to test or reproduce

Test some buttons of the app, if they are correct, if they are clickable or if they work as expected.
You don't need to look at them all, choose a few to validate the test.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation update (if none of the other choices apply)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [X] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/Button/index.tsx [hunk 1]
```diff
@@ -2,19 +2,18 @@ import React from 'react';
...
 	title: string;
 	type: string;
-	onPress(): void;
+	onPress: () => void;
 	disabled: boolean;
 	backgroundColor: string;
 	loading: boolean;
-	theme: TSupportedThemes;
 	color: string;
 	fontSize: any;
```
Comment (diegolmello) [inline app/containers/Button/index.tsx:18]: Can you remove these `any`s?

File: app/containers/Button/index.tsx [hunk 2]
```diff
@@ -40,18 +35,21 @@ const styles = StyleSheet.create({
...
-		type: 'primary',
-		onPress: () => alert('It works!'),
-		disabled: false,
-		loading: false
-	};
-
-	render() {
-		const { title, type, onPress, disabled, backgroundColor, color, loading, style, theme, fontSize, styleText, ...otherProps } =
-			this.props;
+const Button = React.memo(
```
Comment (GleidsonDaniel) [inline app/containers/Button/index.tsx:38]: I don't think it's necessary to use memo in this component.

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: app/containers/Button/index.tsx [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-	disabled: boolean;
-	backgroundColor: string;
-	loading: boolean;
-	theme: TSupportedThemes;
-	color: string;
-	fontSize: any;
-	style: any;
-	styleText?: any;
-	testID: string;
+	onPress: () => void;
```
Comment (GleidsonDaniel) [inline app/containers/Button/index.tsx:12]: Use `Function` to create a generic.

**Review by GleidsonDaniel — COMMENTED** [review]
You could create tests using RNTL and update the storybook stories?

In the tasks of migrating components to hooks we are adding these details.

File: app/containers/Button/index.tsx (hunk 3 above)
Comment (try-catch-stack) [reply app/containers/Button/index.tsx:12]: Using `Function` envokes this error `Interface 'IButtonProps' incorrectly extends interface 'PlatformTouchableProps'.
  Types of property 'onPress' are incompatible.
    Type 'Function' is not assignable to type '(event: GestureResponderEvent) => void'.`
Am I missing something?

**Review by GleidsonDaniel — APPROVED** [review]

**Comment by GleidsonDaniel** [comment]
@try-catch-stack Can you resolve conflicts?
Then you can squash and merge.

**Pull request merged**